### PR TITLE
[GEOT-7571] Support INSERT/UPDATE on PostGIS json/jsonb columns

### PR DESCRIPTION
**Pull request by awaterme**
[GEOT-7571](https://osgeo-org.atlassian.net/browse/GEOT-7571) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=geotools&utm_campaign=badge_info)

Support for INSERT/UPDATE on PostGIS json/jsonb columns.

The change affects the core of the JDBC code. So I have tried out some variants and found this solution the least intrusive. I have run various online tests to make sure not to break things (PostGIS, Oracle, SQL Server, DB2,  HANA, mySQL). Also paid attention to GeoPackage and H2-DB.

# Checklist

- [x] I have read the [contribution guidelines](https://github.com/geotools/geotools/blob/main/CONTRIBUTING.md).
- [x] I have sent a [Contribution Licence Agreement](https://docs.geotools.org/latest/developer/procedures/contribution_license.html) (not required for small changes, e.g., fixing typos in documentation).
- [x] First PR targets the `main` branch (backports managed later; ignore for branch specific issues).
- [x] Avoid [Java 9+ split packages](http://tutorials.jenkov.com/java/modules.html#split-packages-not-allowed).
- [x] All the build checks are green ([see automated QA checks](https://docs.geotools.org/latest/developer/conventions/code/qa.html)).

For core and extension modules:

- [x] New unit tests have been added covering the changes.
- [ ] [Documentation](https://github.com/geotools/geotools/tree/main/docs) has been updated (if change is visible to end users).
- [x] There is an issue in [GeoTools Jira](https://osgeo-org.atlassian.net/projects/GEOT) (except for changes not visible to end users). 
- [x] Commit message(s) must be in the form ``[GEOT-XYZW] Title of the Jira ticket``.
- [x] Bug fixes and small new features are presented as a single commit.
- [x] The commit targets a single objective (if multiple focuses cannot be avoided, each one is in its own commit, and has a separate ticket describing it).

Thank you for considering this!

Best regards,
Andreas

### CODE REVIEW

File: modules/library/jdbc/src/main/java/org/geotools/jdbc/PreparedStatementSQLDialect.java [hunk 1]
```diff
@@ -119,16 +119,23 @@ public abstract void setGeometryValue(
...
      * @param cx The database connection.
      */
     public void setValue(
-            Object value, Class binding, PreparedStatement ps, int column, Connection cx)
+            Object value,
+            Class<?> binding,
+            AttributeDescriptor att,
+            PreparedStatement ps,
+            int column,
+            Connection cx)
```
Comment (aaime) [inline modules/library/jdbc/src/main/java/org/geotools/jdbc/PreparedStatementSQLDialect.java:134]: This is an API change in an existing method that might have been overridden in implementations outside of GeoTools code base. Can you talk about this change in ``docs/user/welcome/upgrade.rst`` for the 32.x series?

As an API break, this bit will prevent backport to the stable/maintenance series.

File: modules/plugin/jdbc/jdbc-postgis/src/test/java/org/geotools/data/postgis/PostGISTestSetup.java [hunk 2]
```diff
@@ -50,11 +51,16 @@ public boolean isPgsqlVersionGreaterThanEqualTo(Version v) {
...
+        SQLDialect sqlDialect = dataStore.getSQLDialect();
+        if (sqlDialect instanceof PostGISDialect) {
+            PostGISDialect d = (PostGISDialect) sqlDialect;
+            // the unit tests assume a non loose behaviour
+            d.setLooseBBOXEnabled(false);
+            d.setEstimatedExtentsEnabled(false);
+        } else {
+            PostGISPSDialect d = (PostGISPSDialect) sqlDialect;
+            d.setLooseBBOXEnabled(false);
+        }
```
Comment (aaime) [inline modules/plugin/jdbc/jdbc-postgis/src/test/java/org/geotools/data/postgis/PostGISTestSetup.java:63]: This change appears to be unrelated, can you describe why it was done?

**Review by aaime — CHANGES_REQUESTED** [review]
Overall looks good, see comments below

File: modules/library/jdbc/src/main/java/org/geotools/jdbc/PreparedStatementSQLDialect.java (hunk 1 above)
Comment (awaterme) [reply modules/library/jdbc/src/main/java/org/geotools/jdbc/PreparedStatementSQLDialect.java:134]: Yes, I'm happy to do that.

File: modules/plugin/jdbc/jdbc-postgis/src/test/java/org/geotools/data/postgis/PostGISTestSetup.java (hunk 2 above)
Comment (awaterme) [reply modules/plugin/jdbc/jdbc-postgis/src/test/java/org/geotools/data/postgis/PostGISTestSetup.java:63]: It was not unrelated. But I found a better solution which is more limited to the affected tests. I will push that instead.

**Comment by awaterme** [comment]
Hi Andrea,
I have replaced previous commit as mentioned above.
Thank you for considering this!
Best regards, Andreas

**Comment by awaterme** [comment]
Db2 online test failure reason:
_Error response from daemon: received unexpected HTTP status: 503 Service Unavailable_

I suppose a further manual run will show a green result.
